src/SourceTerms/Deleptonization/CPU_Src_Deleptonization.cpp

```diff
@@ -282,6 +302,14 @@ static void Src_Deleptonization( real fluid[], const real B[],
 
    } // if ( Del_Ye < (real)0.0 )
 
+
+// update temperature initial guess
+#  ifdef TEMP_IG
+   fluid[TEMP_IG] = Hydro_Con2Temp( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], fluid[ENGY], fluid+NCOMP_FLUID,
+                                    false, 0.0, Emag, EoS->DensEint2Temp_FuncPtr,
+                                    EoS->AuxArrayDevPtr_Flt, EoS->AuxArrayDevPtr_Int, EoS->Table );
+#  endif
```

Comment:
Why do we need to update `TEMP_IG` if there is no changes in the Ye, entropy, and energy?